pkg/instancetype/instancetype.go

```diff
@@ -932,6 +934,29 @@ func validateCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.Vir
 	return conflicts
 }
 
+func applyNodeSelector(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
+	if vmiSpec.NodeSelector != nil {
+		return Conflicts{field.Child("nodeSelector")}
+	}
+
+	vmiSpec.NodeSelector = make(map[string]string, len(instancetypeSpec.NodeSelector))
```

Comment:
supernit - but a TODO here to move to the new `map` package to copy these values in would be appreciated.

https://pkg.go.dev/maps@master